localads/campaign_serialization.cpp

```diff
@@ -0,0 +1,84 @@
+#include "localads/campaign_serialization.hpp"
+
+#include "coding/varint.hpp"
+#include "coding/byte_stream.hpp"
+
+#include <vector>
+#include <type_traits>
+#include <cstdint>
+
+
+namespace
+{
+enum class Version
+{
+  unknown = -1,
+  v1 = 0,  // March 2017 (Store featrues ids as varints, use two bytes for
+           // both days before expiration and icon id.)
+  latest = v1
+};
+
+template<typename ByteStream, typename T>
+void Write(ByteStream & s, T t)
+{
+  s.Write(&t, sizeof(T));
+}
+
+template<typename T, typename ByteStream>
+T Read(ByteStream & s)
+{
+  T t;
+  s.Read(static_cast<void*>(&t), sizeof(t));
+  return t;
+}
+
+template<typename Integral,
+         typename ByteStream,
+         typename std::enable_if<std::is_integral<Integral>::value, void*>::type = nullptr>
+std::vector<Integral> ReadData(ByteStream & s, size_t chunksNumber)
```

Comment:
I suggest you to implement a version of varint.hpp:ReadVarUint64Array for a Source.